src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -421,6 +426,12 @@ static StoragePtr create(const StorageFactory::Arguments & args)
                     "No replica name in config" + getMergeTreeVerboseHelp(is_extended_storage_def), ErrorCodes::NO_REPLICA_NAME_GIVEN);
             ++arg_num;
         }
+        else if (has_replicated_default_args && !has_arguments)
+        {
+            zookeeper_path
+                = args.context.getSettingsRef().default_replica_path.value + "/" + args.table_id.database_name + "." + args.table_id.table_name;
```

Comment:
We have implicit `{database}` and `{table}` macros, so user can specify something like `/clickhouse/tables/{database}/{table}/{shard}/` as default path. I'm not sure if we need append database and table name to default path.

Reply:
Using {database} and {table} macros in ZooKeeper path lead  to serious issues with table renaming.

How it will distinguish the situation "new table" vs "joining replication of existing table" for atomic case? 

If we will start adding some uniqueness automatically for non Atomic database - the same problem will pop up. 


Not sure if it should be implemented